File: BLH.ApproveIQ.Portal/ClientApp/src/app/data/data/b-l-h/entities/invoice-approval.ts
import { AuditableEntity } from '../../../b-l-h/primitives/auditable-entity';
import { Invoice } from './invoice';
import { User } from './user';

export class InvoiceApproval extends AuditableEntity {
    invoiceId: string;
    fromUserId: string;
    toUserId: string;
    fromStatus: string;
    toStatus: string;
    comments: string;
    action: string;
    invoice = new Invoice();
    fromUser = new User();
    toUser = new User();
}

File: BLH.ApproveIQ.Domain/Entities/InvoiceApproval.cs
using BLH.ApproveIQ.Domain.Primitives;

namespace BLH.ApproveIQ.Domain.Entities;

public class InvoiceApproval : AuditableEntity
{
    public Guid InvoiceId { get; set; }
    public Guid FromUserId { get; set; }        // Who assigned/reassigned (could be system for initial)
    public Guid ToUserId { get; set; }          // Who it was assigned to
    public string FromStatus { get; set; } = string.Empty;  // Previous status
    public string ToStatus { get; set; } = string.Empty;    // New status
    public string? Comments { get; set; }       // Optional comments about the action
    public string Action { get; set; } = string.Empty;      // "Assigned", "Approved", "Rejected", "Reassigned"

    // Navigation properties
    public Invoice Invoice { get; set; } = null!;
    public User FromUser { get; set; } = null!;
    public User ToUser { get; set; } = null!;
}
